### PR TITLE
Adjust on-boarding illustrations position and scale

### LINKED ISSUE #20518: Adjust onboarding illustrations position for different screens
# Bug Report

## Problem 

The "Generating keys" illustration position differs from the designs on many screens. 
We should adjust it to match the design. 
Discussion here: https://github.com/status-im/status-mobile/pull/20506#issuecomment-2180953655

Below you can see the comparison of how the illustration is placed on different screens on real devices _vs._ how it looks in Figma.

[Design:](https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=12860-105360&t=HyTaEQLfBQFQq4g5-4)

Screenshot 2024-06-20 at 16 58 11

Pixel 7a:

Screenshot 2024-06-20 at 16 54 56

iPhone SE:

Screenshot 2024-06-20 at 16 55 31

iPhone 12 mini:

Screenshot 2024-06-20 at 16 55 58

Huawei P40 lite:

Screenshot 2024-06-20 at 16 56 18

---

The `Enable biometrics` illustration also has the wrong size on iPhone SE

Screenshot 2024-06-24 at 16 18 54

---

Bad quality and incorrect position of the illustration on the intro screen on Android devices - it looks a bit squeezed vertically. 

Screenshot 2024-06-26 at 14 10 05

--- 

Incorrect position of the new illustration on the intro screen on iOS devices - it should not be cut at the bottom.
Checked on iPhone SE and 12 mini

Screenshot 2024-06-26 at 14 12 27

Design:

Screenshot 2024-06-26 at 14 13 39

### Reproduction

1. Install Status
2. Open the intro screen and check the illustrations
3. Go to the account creation flow
4. Enter your name
5. Create a password
6. Check the illustration position on the `Enable biometrics` screen
7. Check the illustration position on the next `Generating keys` screen

### Additional Information

- Status version: nightly 20/06
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/20518

### Summary & Testing Note

Making illustrations work for different screen sizes is complicated and I think it's tough to replicate Figma designs. When the aspect ratio of the image doesn't match with the image view due to different screen sizes then we only have two choices either crop image or stretch. In the PR I opted for stretching, as it seems less broken, especially in the "enable biometric" screen where stretching is negligible. Apart from that, I introduced other improvements by getting rid of hard codings and uses of absolute layouts.

1. [#20518](https://github.com/status-im/status-mobile/issues/20518) mentions three screens, Generating Keys & Enable Bio-metric screen should be fixed now.
2. The intro screen is more problematic. The subtitle line on top of the intro takes two lines in a few devices (due to size). Also, implementation is complex as we separate the carousel, background, etc, and over-rely on absolute layout and hard codings in each component. To fix the intro screen, I migrated screens to a relative layout and also made a few other changes. So it should be somewhat fixed.

So basically PR doesn't fix screens completely but improves them. Please let me know if this is not the case.

PR also modifies the bottom action quo component, please lightly test other uses to make sure they are not broken.
ex: (quo->drawers->bottom-actions)

status: ready

### CODE REVIEW

File: src/status_im/contexts/onboarding/common/carousel/view.cljs [hunk 1]
```diff
@@ -64,10 +63,10 @@
           :swipeable (animation/composed-gestures progress paused? is-dragging? drag-amount)
           :tappable  (animation/tap-gesture progress paused?)
           nil)]
-    [:<>
+    [rn/view {:style {:flex 1}}
      [gesture/gesture-detector {:gesture identified-gesture}
-      [container-view {:style (style/carousel-container carousel-left animate?)}
-       (for [index (range 2)]
+      [container-view {:style (style/carousel-container carousel-left animate? (* 4 window-width))}
+       (for [index (range 1)]
```
Comment (Parveshdhull) [inline src/status_im/contexts/onboarding/common/carousel/view.cljs:69]: No need to have two layers anymore, infinite scroll is broken.
https://github.com/status-im/status-mobile/issues/20747

**Review by ilmotta — APPROVED** [review]
Looks better overall @Parveshdhull, thank you! I tested on an Galaxy Tab S7, Galaxy A71 and iPhone 14 Pro Max.

It's a tough problem to solve. Since a large segment of users use Android, and that screen specs vary wildly, I think in general stretching images is a broken solution and that we will need to revisit this at some point.

**Review by jo-mut — APPROVED** [review]

**Comment by qoqobolo** [comment]
@Parveshdhull thank for this fix, the illustrations do look better now!
But I have two questions:

### Q1: Low quality of the illustrations on Android

I am aware that we compressed images on purpose to reduce the app size. But do we actually agree with such bad-looking first illustrations that the user sees in the app?
Most likely this is not in the scope of this PR, although I mentioned it in the issue, but maybe you and @ilmotta have some thoughts on this matter? 
Should we report this separately and reconsider the quality of onboarding illustrations on Android?

Screenshot 2024-07-16 at 13 58 06

### Q2: Stretched `Enable biometrics` illustration on Pixel 7a

Check the video please, there is PR on the left and nightly on the right:

https://github.com/user-attachments/assets/69c5b84a-ed70-470a-8094-4e9958f648cb

To me, the illustration looks better in nightly (on the right). It's a device-specific case only reproducible on Pixel 7a, on other devices it looks good.
What do you think about it? Can we keep the illustration as it is on nightly only for Pixel?
If not, then maybe we should sacrifice the `Enable biometrics` illustration on iPhone SE, since we are more focused on Android users, and on other iOS devices it looks okay?
@ilmotta your opinion would also be appreciated.
Thank you!

**Comment by Parveshdhull** [comment]
Thank you very much @qoqobolo for testing the PR.

> Q. 1. Should we report this separately and reconsider the quality of onboarding illustrations on Android?

Yes, better to report this separately. This will require some discussion and exploration to avoid increasing app size while maintaining quality.

> Q.2 Stretched Enable biometrics illustration on Pixel 7a

Please let me know if the enable biometric screen looks better now.

**Comment by ilmotta** [comment]
@qoqobolo @Parveshdhull I agree with your comments. In a future release we will need to revisit this approach of using images that occupy the full device width or height. For 2.30 it's acceptable with the improvements brought by this PR.

There will always be devices where users will see deformed images because we stretch them to fit. The onboarding carousel is a great example of the problem. If we want to solve this problem once and for all, I think we will need new designs and new illustrations that don't rely on the device's width. For example, designs that will look good even with cropped images, or designs that assume a large padding around images, so that we can accommodate for most devices. Another solution would be to have more images to accommodate for more devices, but I think this is not where we want to go because the APK is already close to 100MB.

**Comment by qoqobolo** [comment]
Thanks again @Parveshdhull and @ilmotta, agree on all points.
PR can be merged.
